packages/material-ui/src/styles/withStyles.js

```diff
@@ -93,6 +93,9 @@ const withStyles = (stylesOrCreator, options = {}) => Component => {
       this.theme = listenToTheme ? themeListener.initial(context) || getDefaultTheme() : noopTheme;
 
       this.attach(this.theme);
+      this.classes = {};
+      this.lastClassesProp = {};
+      this.lastRenderedClasses = {};
```

Comment:
Cache for the finalized classes value.
Cache for the last used classes prop pointer (to determine if it changed).
Cache for the last used rendered classes pointer (to determine if it changed).